consensus/types/src/sync_committee_contribution.rs

```diff
@@ -0,0 +1,115 @@
+use super::{AggregateSignature, ChainSpec, Domain, EthSpec, Fork, SecretKey, SignedRoot};
+use crate::attestation::SlotData;
+use crate::{test_utils::TestRandom, BitVector, Hash256, Slot, SyncCommitteeSignature};
+use safe_arith::ArithError;
+use serde_derive::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use test_random_derive::TestRandom;
+use tree_hash_derive::TreeHash;
+
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    SszTypesError(ssz_types::Error),
+    AlreadySigned(usize),
+    SubnetCountIsZero(ArithError),
+}
+
+/// Details an attestation that can be slashable.
+///
+/// Spec v1.1.0
+#[cfg_attr(feature = "arbitrary-fuzz", derive(arbitrary::Arbitrary))]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode, TreeHash, TestRandom)]
+#[serde(bound = "T: EthSpec")]
+pub struct SyncCommitteeContribution<T: EthSpec> {
+    pub slot: Slot,
+    pub beacon_block_root: Hash256,
+    pub subcommittee_index: u64,
+    pub aggregation_bits: BitVector<T::SyncCommitteeSubnetSize>,
+    pub signature: AggregateSignature,
+}
+
+impl<T: EthSpec> SyncCommitteeContribution<T> {
+    pub fn from_signature(
+        signature: SyncCommitteeSignature,
+        subnet_id: u64,
+        subcommittee_index: u64,
```

Comment:
These two parameters are confusingly named because the spec uses both of them to refer to the same concept, e.g. https://github.com/ethereum/eth2.0-specs/blob/dev/specs/altair/validator.md#aggregation-selection

The terminology for the 2nd parameter (i.e. the position of the validator within the committee) seems to be settling on `validator_sync_committee_index`. I'll make this change to my branch and see if I can push to your branch.